nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/state/providers/zookeeper/ZooKeeperStateProvider.java

```diff
@@ -133,20 +147,52 @@ public ZooKeeperStateProvider() {
         return properties;
     }
 
-
     @Override
     public synchronized void init(final StateProviderInitializationContext context) {
         connectionString = context.getProperty(CONNECTION_STRING).getValue();
         rootNode = context.getProperty(ROOT_NODE).getValue();
         timeoutMillis = context.getProperty(SESSION_TIMEOUT).asTimePeriod(TimeUnit.MILLISECONDS).intValue();
 
+        final Properties stateProviderProperties = new Properties();
+        stateProviderProperties.setProperty(NiFiProperties.ZOOKEEPER_SESSION_TIMEOUT, String.valueOf(timeoutMillis));
+        stateProviderProperties.setProperty(NiFiProperties.ZOOKEEPER_CONNECT_TIMEOUT, String.valueOf(timeoutMillis));
+        stateProviderProperties.setProperty(NiFiProperties.ZOOKEEPER_ROOT_NODE, rootNode);
+        stateProviderProperties.setProperty(NiFiProperties.ZOOKEEPER_CONNECT_STRING, connectionString);
+
+        zooKeeperClientConfig = ZooKeeperClientConfig.createConfig(combineProperties(nifiProperties, stateProviderProperties));
```

Comment:
If you're happy, I might leave the existing properties as-is so that the configuration for the file is still backwards compatible, and I'll simply enable the required=true for the property. I've fixed a small issue with the StandardStateManagerProvider.java validator execution order so startup fails if the property is not present. This will make the error log message relevant again, and uses the connect string validator.

We could make these properties not required in the state-management.xml and instead inherited from nifi.properties, but for now I guess I'll stick with what's expected and keep them in this file. If you think we should instead inherit from nifi.properties by default I can do that too so let me know what you think.

Reply:
Apparently the order of the initialization -> validator was correct so I reverted that back and stopped the properties from being used in initialization of the ZooKeeperStateProvider to fix the issue of the validator not catching empty/missing properties.